src/magicbox/SuggestionsManager.ts

```diff
@@ -514,7 +547,66 @@ export class SuggestionsManager {
   }
 
   private htmlElementIsSuggestion(selected: HTMLElement) {
-    const omniboxSelectables = $$(this.element).findAll(`.${this.options.selectableClass}`);
+    const omniboxSelectables = $$(this.element).findAll(`.${this.options.suggestionClass}`);
     return indexOf(omniboxSelectables, selected) > -1;
   }
+
+  private getSearchResultPreviewsQuery(suggestion: HTMLElement) {
+    const populateEventArgs: IPopulateSearchResultPreviewsEventArgs = {
+      suggestionText: suggestion.innerText,
+      previewsQuery: null
+    };
+    $$(this.root).trigger(SuggestionsManagerEvents.PopulateSearchResultPreviews, populateEventArgs);
+    return populateEventArgs.previewsQuery;
+  }
```

Comment:
This is where `QuerySuggestPreview` is called to feed search result previews. Other components could also respond to it, but receiving an array of subpromises would involve making another function like `this.mergeSuggestions`.  This could be changed once `QueryProcessor` is added.